packages/flutter/lib/src/rendering/table.dart

```diff
@@ -586,124 +533,54 @@ class RenderTable extends RenderBox {
       child.parentData = TableCellParentData();
   }
 
-  /// Replaces the children of this table with the given cells.
+  /// Insert child into this render object's child list after the given child.
   ///
-  /// The cells are divided into the specified number of columns before
-  /// replacing the existing children.
-  ///
-  /// If the new cells contain any existing children of the table, those
-  /// children are simply moved to their new location in the table rather than
-  /// removed from the table and re-added.
-  void setFlatChildren(int columns, List<RenderBox?> cells) {
-    if (cells == _children && columns == _columns)
-      return;
-    assert(columns >= 0);
-    // consider the case of a newly empty table
-    if (columns == 0 || cells.isEmpty) {
-      assert(cells == null || cells.isEmpty);
-      _columns = columns;
-      if (_children.isEmpty) {
-        assert(_rows == 0);
-        return;
-      }
-      for (final RenderBox? oldChild in _children) {
-        if (oldChild != null)
-          dropChild(oldChild);
-      }
-      _rows = 0;
-      _children.clear();
-      markNeedsLayout();
-      return;
-    }
-    assert(cells != null);
-    assert(cells.length % columns == 0);
-    // fill a set with the cells that are moving (it's important not
-    // to dropChild a child that's remaining with us, because that
-    // would clear their parentData field)
-    final Set<RenderBox> lostChildren = HashSet<RenderBox>();
-    for (int y = 0; y < _rows; y += 1) {
-      for (int x = 0; x < _columns; x += 1) {
-        final int xyOld = x + y * _columns;
-        final int xyNew = x + y * columns;
-        if (_children[xyOld] != null && (x >= columns || xyNew >= cells.length || _children[xyOld] != cells[xyNew]))
-          lostChildren.add(_children[xyOld]!);
-      }
+  /// If `after` is null, then this inserts the child at the start of the list.
```

Comment:
Or can we change this to an API that's more explicit about the actual child model, maybe something like:

```dart
void insertChild(RenderBox child, {required int row, required int column});
```

Since the element has access to the index of the child, could it use that to calculate what column and row the child needs to be in?

Reply:
Updated the comment below.

Reply:
Now `_children` is variable length, use `row` or `column` is not suitable.